Oracle/med_admin.sql

```diff
@@ -0,0 +1,140 @@
+-- HELPER FUNCTIONS AND PROCEDURES
+--------------------------------------------------------------------------------
+--These helper functions also exist in PCORNetInit.sql.  They are reproduced
+--here to make the med_admin script an independent luigi job.
+--TODO: consolidate helpers in a single, table independent job.
+create or replace PROCEDURE GATHER_TABLE_STATS(table_name VARCHAR2) AS
+  BEGIN
+  DBMS_STATS.GATHER_TABLE_STATS (
+          ownname => 'PCORNET_CDM', -- This doesn't work as a parameter for some reason.
+          tabname => table_name,
+          estimate_percent => 50, -- Percentage picked somewhat arbitrarily
+          cascade => TRUE,
+          degree => 16
+          );
+END GATHER_TABLE_STATS;
+/
+create or replace PROCEDURE PMN_DROPSQL(sqlstring VARCHAR2) AS
+  BEGIN
+      EXECUTE IMMEDIATE sqlstring;
+  EXCEPTION
+      WHEN OTHERS THEN NULL;
+END PMN_DROPSQL;
+/
+--------------------------------------------------------------------------------
+-- MED_ADMIN
+--------------------------------------------------------------------------------
+BEGIN
+PMN_DROPSQL('DROP TABLE med_admin');
+END;
+/
+CREATE TABLE med_admin(
+    MEDADMINID varchar(50) primary key,
+    PATID varchar(50) NOT NULL,
+    ENCOUNTERID varchar(50) NULL,
+    PRESCRIBINGID varchar(50) NULL,
+    MEDADMIN_PROVIDERID varchar(50) NULL,
+    MEDADMIN_START_DATE date NOT NULL,
+    MEDADMIN_START_TIME varchar(5) NULL,
+    MEDADMIN_STOP_DATE date NULL,
+    MEDADMIN_STOP_TIME varchar(5) NULL,
+    MEDADMIN_TYPE varchar(2) NULL,
+    MEDADMIN_CODE varchar(50) NULL,
+    MEDADMIN_DOSE_ADMIN NUMBER(18, 2) NULL, -- (8,0)
+    MEDADMIN_DOSE_ADMIN_UNIT varchar(50) NULL,
+    MEDADMIN_ROUTE varchar(50) NULL,
+    MEDADMIN_SOURCE varchar(2) NULL,
+    RAW_MEDADMIN_MED_NAME varchar(50) NULL,
+    RAW_MEDADMIN_CODE varchar(50) NULL,
+    RAW_MEDADMIN_DOSE_ADMIN varchar(50) NULL,
+    RAW_MEDADMIN_DOSE_ADMIN_UNIT varchar(50) NULL,
+    RAW_MEDADMIN_ROUTE varchar(50) NULL
+)
+/
+BEGIN
+PMN_DROPSQL('DROP sequence med_admin_seq');
+END;
+/
+create sequence med_admin_seq
+/
+create or replace trigger med_admin_trg
+before insert on med_admin
+for each row
+begin
+  select med_admin_seq.nextval into :new.MEDADMINID from dual;
+end;
+/
+create or replace procedure PCORNetMedAdmin as
+begin
+
+PMN_DROPSQL('drop index med_admin_idx');
+
+execute immediate 'truncate table med_admin';
+
+insert into med_admin(patid, encounterid, medadmin_providerid, medadmin_start_date, medadmin_start_time, medadmin_stop_date,
+medadmin_stop_time, medadmin_type, medadmin_code, medadmin_dose_admin, medadmin_dose_admin_unit, medadmin_source, raw_medadmin_med_name,
+raw_medadmin_code, raw_medadmin_dose_admin, raw_medadmin_dose_admin_unit, raw_medadmin_route)
+
+with med_start as (
+    select patient_num, encounter_num, provider_id, start_date, end_date, concept_cd, modifier_cd, instance_num
+    from BLUEHERONDATA.observation_fact
+    where modifier_cd = 'MedObs|MAR:New Bag'
```

Comment:
med_admin.sql is the major piece of functional code for the first deliverable.

As mentioned in a prior review, these modifier should be moved to a configuration file.  Todo, check with Matt for existing code, port it from HERON?